graph/pull_v2.go

```diff
@@ -243,6 +199,16 @@ func (p *v2Puller) pullV2Tag(out io.Writer, tag, taggedName string) (verified bo
 	var layerIDs []string
 	defer func() {
 		p.graph.Release(p.sessionID, layerIDs...)
+
+		for _, d := range downloads {
+			p.poolRemoveWithError("pull", d.poolKey, err)
+			if d.tmpFile != nil {
```

Comment:
I agree with the suggestion in general, but in this particular case I'd rather leave it as-is, in case we add other cleanup tasks to this loop in the future (which would be skipped by the continue).